### PR TITLE
added `--logfile` option

### PR DESCRIPTION
**Pull request by Joe-Merten**
as discussed in https://github.com/npat-efault/picocom/issues/24

### CODE REVIEW

File: picocom.c [hunk 1]
```diff
@@ -1202,6 +1213,9 @@ loop(void)
 			} while ( n < 0 && errno == EINTR );
 			if ( n <= 0 )
 				fatal("write to term failed: %s", strerror(errno));
+			if ( opts.lecho && opts.log_filename )
+				if ( writen_ni(log_fd, tty_q.buff, sz) < sz )
```
Comment (npat-efault) [inline picocom.c]: Small bug: I think it should be:

    writen_ni(log_fd, tty_q.buff, n) < n

Comment (Joe-Merten) [inline picocom.c]: Talking about picocom.c:1194, I think that `buff_rd` is correct. When I change this to `tty_q.buff`, then I got wrong content in the logfile.

Comment (npat-efault) [inline picocom.c]: No, I'm talking about picocom.c:1217, and the problem is not `tty_q.buff`, but `n` instead of `sz`.

Comment (Joe-Merten) [inline picocom.c]: Yep, checked the logic of this code fragment again, now being agree with you.

**Pull request merged**